server/mongo_data_access.go

```diff
@@ -465,6 +467,23 @@ func (dal *mongoDataAccessLayer) Search(baseURL url.URL, searchQuery search.Quer
 	return &bundle, nil
 }
 
+// testUsesChainedSearch tests if a search.Query has at least one SearchParam
+// that is a ReferenceParam of type ChainedQueryReference; therefore the query
+// uses chained search.
+func testUsesChainedSearch(q search.Query) bool {
```

Comment:
I'd prefer the function name not start with `test`, as this is the convention for function names that are _unit tests_.  I think `func usesChainedSearch(q search.Query) bool` would suffice.